### PR TITLE
Malaysia Calendar

### PR DESCRIPTION
**Pull request by peterhbromley**

### CODE REVIEW

File: trading_calendars/exchange_calendar_xkls.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+            NuzulAlQuran,
+            EidAlFitrDay1,
+            EidAlFitrDay2,
+            EidAlAdha,
+            Muharram,
+            MuhammadBirthday,
+            Deepavali,
+            Thaipusam,
+            WesakDay,
+        ))
```
[the remark this answers is not in the document]
Comment (peterhbromley) [reply trading_calendars/exchange_calendar_xkls.py]: The days depend on the current head of state (Yang di-Pertuan Agong means head of state) and both holidays do not happen every year (I think they maybe only occur when there is a new one?)

File: trading_calendars/xkls_holidays.py [hunk 2]
```diff
@@ -0,0 +1,512 @@
...
+    '2043-02-22',
+    '2044-02-11',
+    '2045-01-30',
+    '2046-01-19',
+    '2047-01-08',
+    '2047-12-29',
+    '2048-12-18',
+])
+
+# Malaysia's observances of Eid al-Fitr
```
[the remark this answers is not in the document]
Comment (peterhbromley) [reply trading_calendars/xkls_holidays.py]: Not sure exactly why, but it likely has to do with the fact that Malaysia and Turkey are far apart geographically so maybe observations of Lunar events differ by a day in some cases?

File: trading_calendars/xkls_holidays.py [hunk 3]
```diff
@@ -0,0 +1,512 @@
...
+    '2004-01-21',  # Chinese New Year extra day
+    '2003-11-24',  # Eid al-Fitr extra day
+    '2003-02-04',  # Chinese New Year extra day
+    '2003-01-31',  # Chinese New Year extra day
+    '2002-04-25',  # Installation of Yang di-Pertuan Agong
+    '2002-02-11',  # Chinese New Year extra day
+])
+
+
+def manual_sunday_to_monday(holidays):
```
[the remark this answers is not in the document]
Comment (peterhbromley) [reply trading_calendars/xkls_holidays.py]: Yes, that does work!  Thanks for noticing that, made the change

File: trading_calendars/exchange_calendar_xkls.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mrkgoh) [reply trading_calendars/exchange_calendar_xkls.py:121]: I am from Malaysia. The sultan rotates every 5 years. Hence their birthdays changes too.

**Pull request merged**